src/sentry/analytics/map.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any, Sequence
+
+from . import Attribute
+from .utils import get_data
+
+
+class Map(Attribute):
+    def __init__(
+        self, name: str, attributes: Sequence[Attribute] | None = None, required: bool = True
+    ) -> None:
+        super().__init__(name, str, required)
+        self.attributes = attributes or ()
+
+    def extract(self, value: dict[str, Any] | Any | None) -> Mapping[str, Any] | None:
+        """
+        If passed a non dictionary we assume we can pull attributes from it.
+
+        This will hard error in some situations if you're passing a bad type
+        (like an int).
+        """
+        if value is None:
+            return value
+
+        if isinstance(value, dict):
+            # Ensure we don't mutate the original. We do not need to deepcopy;
+            # if it recurses into another Map it will once again copy itself.
```

Comment:
You're aware that if the dictionary has a list or some other complex object in it, that object will not be copied? The comment is a little unclear here, I can't tell if you're saying `copy` will recurse (which it won't) or `extract` recurses.

Reply:
I just copied this class to a file so I don't know what the intention was for this comment. I'll leave a note in the code here.